contracts/LinearVesting.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title LinearVesting
+ * @dev
+ */
+contract LinearVesting is Ownable {
+    IERC20 token;
+    uint256 public start; // when the vesting starts
+    uint256 public cliff; // period in seconds
+    uint256 public duration; // vesting duration
+    address receiver; // recipient
+    uint256 amount; // number of tokens to be sent
+    uint256 totalAmount; // total amount of tokens locked in the contract
+
+    // add event
+
+    constructor(address _token, uint256 _duration, uint256 _cliff) {
+        require(_duration > 0, "Duration must be positive");
+        require(
+            _duration >= _cliff,
+            "Cliff cannot be greater than the vesting duration"
+        );
```

Comment:
Cliff can be defined differently : either the vesting starts at the end of the cliff, or it starts, but then once the cliff is finished there is a high release of tokens on a given day.

Reply:
I've corrected the vesting logic error : no tokens are released until the cliff period has elapsed.